src/Mod/Sketcher/Gui/DrawSketchHandlerScale.h

```diff
@@ -0,0 +1,529 @@
+            return;
+        }
+
+        if (fabs(scaleFactor) < Precision::Confusion()) {
+            return;
+        }
+
+        for (auto& geoId : listOfGeoIds) {
+            const Part::Geometry* pGeo = Obj->getGeometry(geoId);
+            auto geoUniquePtr = std::unique_ptr<Part::Geometry>(pGeo->copy());
+            Part::Geometry* geo = geoUniquePtr.get();
+
+            if (isCircle(*geo)) {
+                auto* circle = static_cast<Part::GeomCircle*>(geo);  // NOLINT
+                circle->setRadius(circle->getRadius() * scaleFactor);
+                circle->setCenter(getScaledPoint(circle->getCenter(), referencePoint, scaleFactor));
+            }
+            else if (isArcOfCircle(*geo)) {
+                auto* arcOfCircle = static_cast<Part::GeomArcOfCircle*>(geo);  // NOLINT
+                arcOfCircle->setRadius(arcOfCircle->getRadius() * scaleFactor);
+                arcOfCircle->setCenter(
+                    getScaledPoint(arcOfCircle->getCenter(), referencePoint, scaleFactor));
+            }
+            else if (isLineSegment(*geo)) {
+                auto* line = static_cast<Part::GeomLineSegment*>(geo);  // NOLINT
+                line->setPoints(getScaledPoint(line->getStartPoint(), referencePoint, scaleFactor),
+                                getScaledPoint(line->getEndPoint(), referencePoint, scaleFactor));
+            }
+            else if (isEllipse(*geo)) {
+                auto* ellipse = static_cast<Part::GeomEllipse*>(geo);  // NOLINT
+                ellipse->setMajorRadius(ellipse->getMajorRadius() * scaleFactor);
+                ellipse->setMinorRadius(ellipse->getMinorRadius() * scaleFactor);
+                ellipse->setCenter(
+                    getScaledPoint(ellipse->getCenter(), referencePoint, scaleFactor));
+            }
+            else if (isArcOfEllipse(*geo)) {
+                auto* arcOfEllipse = static_cast<Part::GeomArcOfEllipse*>(geo);  // NOLINT
+                arcOfEllipse->setMajorRadius(arcOfEllipse->getMajorRadius() * scaleFactor);
+                arcOfEllipse->setMinorRadius(arcOfEllipse->getMinorRadius() * scaleFactor);
+                arcOfEllipse->setCenter(
+                    getScaledPoint(arcOfEllipse->getCenter(), referencePoint, scaleFactor));
+            }
+            else if (isArcOfHyperbola(*geo)) {
+                auto* arcOfHyperbola = static_cast<Part::GeomArcOfHyperbola*>(geo);  // NOLINT
+                arcOfHyperbola->setMajorRadius(arcOfHyperbola->getMajorRadius() * scaleFactor);
+                arcOfHyperbola->setMinorRadius(arcOfHyperbola->getMinorRadius() * scaleFactor);
+                arcOfHyperbola->setCenter(
+                    getScaledPoint(arcOfHyperbola->getCenter(), referencePoint, scaleFactor));
+            }
+            else if (isArcOfParabola(*geo)) {
+                auto* arcOfParabola = static_cast<Part::GeomArcOfParabola*>(geo);  // NOLINT
+                // TODO: Problem with scale parabola end points.

```

Comment:
Problem?

Reply:
@PaddleStroke are we going to leave this for a later PR to resolve?

Reply:
This is now fixed